cmd/crossplane/core/core.go

```diff
@@ -273,13 +275,25 @@ func (c *startCommand) Run(s *runtime.Scheme, log logging.Logger) error { //noli
 		return errors.Wrap(err, "cannot setup API extension controllers")
 	}
 
+	var pr pkgcontroller.PackageRuntime
+	switch c.PackageRuntime {
+	case string(pkgcontroller.PackageRuntimeDeployment):
+		pr = pkgcontroller.PackageRuntimeDeployment
+	case string(pkgcontroller.PackageRuntimeExternal):
+		pr = pkgcontroller.PackageRuntimeExternal
+	default:
+		return errors.Errorf("unsupported package runtime %q, supported runtimes are %q and %q",
+			c.PackageRuntime, pkgcontroller.PackageRuntimeDeployment, pkgcontroller.PackageRuntimeExternal)
+	}
+
 	po := pkgcontroller.Options{
 		Options:         o,
 		Cache:           xpkg.NewFsPackageCache(c.CacheDir, afero.NewOsFs()),
 		Namespace:       c.Namespace,
 		ServiceAccount:  c.ServiceAccount,
 		DefaultRegistry: c.Registry,
 		FetcherOptions:  []xpkg.FetcherOpt{xpkg.WithUserAgent(c.UserAgent)},
+		PackageRuntime:  pr,
```

Comment:
Should (can?) we put package runtime support behind a beta feature flag?

Reply:
This will not have much benefit at this moment, I cannot think of a reason for a user to want to disable the feature.
I prefer to keep this out of feature flagging as the successor of controllerConfig.

Reply:
This PR combined two things I attempted to separate but later [gave up](https://github.com/crossplane/crossplane/pull/4767#issuecomment-1761561210): refactoring of existing manifest building and implementing the DeploymentRuntimeConfig. 

Currently, if we want to keep refactoring but bind `DeploymentRuntimeConfig` to a beta flag, what I would do when it is disabled would be to start with an empty deployment template instead of [reading](https://github.com/crossplane/crossplane/pull/4744/files#diff-84987f336046c313143ebf6fb99ef8883be00dd70ad1645ed3b8235cd24f1b1aR186) it from the referenced `DeploymentRuntimeConfig` which wouldn't cause much difference/value.

If I had taken this into account since the beginning, I could have kept the existing code and done the refactoring and implementation as another alternative hook implementation and plugged it when the feature is enabled. This way, I would see more value in providing a way to keep using the previous implementation in case I introduce some bugs here. However, with the current state, I believe that would require some effort that wouldn't pay off and TBH I am happy that we are getting rid of that previous code sooner than later :)

Reply:
> what I would do when it is disabled would be to start with an empty deployment template instead of reading it from the referenced DeploymentRuntimeConfig which wouldn't cause much difference/value.

This could still be helpful when someone is uncomfortable enabling the flexible/powerful configuration interface we're introducing here. 

https://github.com/crossplane/crossplane/pull/4857